src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/ActivatorUtilitiesTests.cs

```diff
@@ -262,11 +286,19 @@ public void TypeActivatorThrowsOnMultipleMarkedCtors(CreateInstanceFunc createFu
         [MemberData(nameof(CreateInstanceFuncs))]
         public void TypeActivatorThrowsWhenMarkedCtorDoesntAcceptArguments(CreateInstanceFunc createFunc)
         {
-            // Act
-            var exception = Assert.Throws<InvalidOperationException>(() => CreateInstance<ClassWithAmbiguousCtorsAndAttribute>(createFunc, null, 0, "hello"));
+            // Arrange
+            string message = "Constructor marked with ActivatorUtilitiesConstructorAttribute does not accept all given argument types.";
+            var serviceCollection = new TestServiceCollection();
+            var serviceProvider = CreateServiceProvider(serviceCollection);
 
-            // Assert
-            Assert.Equal("Constructor marked with ActivatorUtilitiesConstructorAttribute does not accept all given argument types.", exception.Message);
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() =>
+                CreateInstance<ClassWithAmbiguousCtorsAndAttribute>(createFunc, serviceProvider, 0, "hello"));
+            Assert.Equal(message, exception.Message);
+
+            exception = Assert.Throws<InvalidOperationException>(() =>
+                CreateInstance<ClassWithMultipleAmbiguousCtorsAndAttribute>(createFunc, serviceProvider, 0, "hello"));
```

Comment:
What does `ClassWithMultipleAmbiguousCtorsAndAttribute` mean? How is it different from `ClassWithAmbiguousCtorsAndAttribute`?

Reply:
Removed, they seemed similar for the purpose of the test